### PR TITLE
IoTivity module for IoT.js

### PR DESCRIPTION
**Pull request by abuczak**
This implements an `IoTivity` module for Linux (for now) platform. The API is almost the same as the  IoTivity C API. Our API is documented [here](https://github.com/abuczak/iotjs/blob/IoTivity/docs/api/IoT.js-API-IoTivity.md). 

To build this module download and build dependencies using script `tools/iotivity.sh`.

IoT.js-DCO-1.0-Signed-off-by: Albert Buczak a.buczak@samsung.com

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,3 +36,4 @@ TAGS
 # config files
 *.config
 !build.config
+!build.iotivity.config
```
Comment (sumin-lim) [inline .gitignore]: 1. Remove this end of line character.
2. We didn't talk about using another config file. It's duplicated. Should this file exist separately? Don't you add it on the build.config file?

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,8 @@
 /src/iotjs_js.c
 /src/iotjs_string_ext.inl.h
 /test/tmp/*
+/iotivity
+/iotivity/*
```
Comment (sumin-lim) [inline .gitignore]: what's the difference between iotivity/* and iotivity?

File: build.iotivity.config [hunk 3]
```diff
@@ -0,0 +1,129 @@
+{
```
Comment (sumin-lim) [inline build.iotivity.config]: There are two way to provide this build option . 
 1. integrate this file on the exist one
 2. make developer use this option as a parameter on the build.
I prefer second one, we can specify with external-static-lib or compile with parameter on using  tools/build.py. Why don't you add this guide on  the API document?

File: samples/iotivity/server_get.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      });
+
+      return iotivity.OCEntityHandlerResult.OC_EH_OK;
+    }
+
+    // By default we error out
+    return iotivity.OCEntityHandlerResult.OC_EH_ERROR;
+  },
+  // This is the same value as server.get.js
+  sampleUri = "/a/iotivity-node-get-sample",
```
Comment (sumin-lim) [inline samples/iotivity/server_get.js]: iotivity-node?

File: samples/iotivity/server_get.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+iotivity.OCSetDeviceInfo({
+  specVersion: "res.1.1.0",
+  dataModelVersions: ["abc.0.0.1"],
+  deviceName: "server.get",
+  types: []
+});
+iotivity.OCSetPlatformInfo({
+  platformID: "server.get.sample",
+  manufacturerName: "iotivity-node"
```
Comment (sumin-lim) [inline samples/iotivity/server_get.js]: iotivity-node?

File: samples/iotivity/server_discoverable.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+iotivity.OCSetDeviceInfo({
+  dataModelVersions: ["test.0.0.1"],
+  deviceName: "server.discoverable",
+  types: []
+});
+
+iotivity.OCSetPlatformInfo({
+  platformID: "server.discoverable.sample",
+  manufacturerName: "iotivity-node",
```
Comment (sumin-lim) [inline samples/iotivity/server_discoverable.js]: iotivity-node?

File: samples/iotivity/server_put.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+iotivity.OCSetDeviceInfo({
+  specVersion: "res.1.1.0",
+  dataModelVersions: ["abc.0.0.1"],
+  deviceName: "server.observable",
+  types: ["oic.d.tv"]
+});
+
+iotivity.OCSetPlatformInfo({
+  platformID: "server.observe.sample",
+  manufacturerName: "iotivity-node"
```
Comment (sumin-lim) [inline samples/iotivity/server_put.js]: iotivity-node?

File: samples/iotivity/client_get.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var intervalId,
+  handleReceptacle = {},
+  sampleUri = "/a/iotivity-node-get-sample",
```
Comment (sumin-lim) [inline samples/iotivity/client_get.js]: iotivity-node?

File: test/run_pass/iotivity-utils.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  },
+  write: function (buffer, totalSize, fp) {
+    return fs.writeSync(fp, buffer, 0, totalSize, null);
+  },
+  unlink: function (path) {
+    fs.unlinkSync(path);
+    return 0;
+  },
+  clean: function() {
+    fs.unlinkSync("oic_svr_db.dat");
```
Comment (sumin-lim) [inline test/run_pass/iotivity-utils.js]: Could you explain about this file? "oic_svr_db.dat" 
When does it create?

File: docs/api/IoT.js-API-IoTivity.md [hunk 10]
```diff
@@ -0,0 +1,613 @@
...
+        resourceUri: "/a/" + uuid,
+        sendVendorSpecificHeaderOptions: []
+      });
+    }
+    return iotivity.OCEntityHandlerResult.OC_EH_OK;
+  }
+);
+```
+
+### More examples are in samples/iotivity/
```
Comment (sumin-lim) [inline docs/api/IoT.js-API-IoTivity.md]: remove ### . it should not be title

File: docs/api/IoT.js-API-IoTivity.md [hunk 11]
```diff
@@ -0,0 +1,613 @@
+### Platform Support
+
+The following shows `IoTivity` module APIs available for each platform.
+
+|  | Linux<br/>(Ubuntu) | Raspbian<br/>(Raspberry Pi) | NuttX<br/>(STM32F4-Discovery) |
+| :---: | :---: | :---: | :---: |
+| iotivity.OCInit | O | O | x |
```
Comment (sumin-lim) [inline docs/api/IoT.js-API-IoTivity.md]: align text on to the left not the middle,center

**Comment by sumin-lim** [comment]
Albert, please close this PR. I'm sending a reason on email.

**Comment by yichoi** [comment]
I'm closing

**Pull request closed without merging**